src/WS2812FX/effects/ColorWipeBaseEffect.cpp

```diff
@@ -0,0 +1,114 @@
+    
+    // Initialize effect-specific state
+    currentColorIndex = EffectHelper::get_random_wheel_index(0, 32);
+    previousColorIndex = EffectHelper::get_random_wheel_index(currentColorIndex, 32);
+    previousPosition = 0;
+    needNewColor = true;
+    isMovingUp = true;
+    
+    return initResult;
+}
+
+void ColorWipeBaseEffect::updateColorIndices(WS2812FX* strip) {
+    if (needNewColor) {
+        previousColorIndex = currentColorIndex;
+        currentColorIndex = EffectHelper::get_random_wheel_index(currentColorIndex, 32);
+        needNewColor = false;
+    }
+}
+
+void ColorWipeBaseEffect::fillWipeColors(WS2812FX* strip, uint16_t position) {
+    auto seg = strip->getSegment();
+    auto runtime = strip->getSegmentRuntime();
+    if (!seg || !runtime) return;
+    
+    // Get colors from palette with current indices
+    CRGB color1 = strip->ColorFromPaletteWithDistribution(
+        *strip->getCurrentPalette(), 
+        currentColorIndex + runtime->baseHue, 
+        seg->targetBrightness, 
+        seg->blendType);
+        
+    CRGB color2 = strip->ColorFromPaletteWithDistribution(
+        *strip->getCurrentPalette(), 
+        previousColorIndex + runtime->baseHue, 
+        seg->targetBrightness, 
+        seg->blendType);
+    
+    // Fill the strip based on direction and position
+    if (isMovingUp) {
+        // Moving up: fill from start to position with color1, rest with color2
+        if (position > runtime->start) {
+            uint16_t fillLength = position - runtime->start;
+            fill_solid(&strip->leds[runtime->start], fillLength, color1);
+        }
+        if (position < runtime->stop) {
+            uint16_t fillLength = runtime->stop - position;
+            fill_solid(&strip->leds[position], fillLength, color2);
+        }
+    } else {
+        // Moving down: fill from position to end with color1, start to position with color2
+        if (position < runtime->stop) {
+            uint16_t fillLength = runtime->stop - position;
+            fill_solid(&strip->leds[position], fillLength, color1);
+        }
+        if (position > runtime->start) {
+            uint16_t fillLength = position - runtime->start;
+            fill_solid(&strip->leds[runtime->start], fillLength, color2);
+        }
+    }
+}
+
+uint16_t ColorWipeBaseEffect::update(WS2812FX* strip) {
+    // Validate strip pointer using helper
+    if (!EffectHelper::validateStripPointer(strip)) {
+        return 1000;
+    }
+    
+    // Access segment data for configuration
+    auto seg = strip->getSegment();
+    auto runtime = strip->getSegmentRuntime();
+    if (!seg || !runtime) {
+        return strip->getStripMinDelay();
+    }
+    
+    // Calculate wipe position using subclass-specific method
+    uint16_t wavePosition = calculateWipePosition(strip, timebase);
+    
+    // Map wave position to LED strip coordinates
+    uint16_t ledPosition = map(wavePosition, 
```

Comment:
we map the 16bit wave position to the leds to apply them to the segment.
However, couldn't we also use a kine of "fractionalFill" with the 16bit values like "drawFractionalBar" functionality but with a full lid starting pixel, full lid width and only the last pixel showing the fraction? This would smooth the animation.

Task:
- check if and how this is feasible.